services/ui-src/src/components/forms/Form.tsx

```diff
@@ -6,15 +6,27 @@ import { yupResolver } from "@hookform/resolvers/yup";
 import { Box } from "@chakra-ui/react";
 import { ReportContext } from "components";
 // utils
-import { formFieldFactory, hydrateFormFields, sortFormErrors } from "utils";
-import { FormJson, FormField } from "types";
+import {
+  formFieldFactory,
+  hydrateFormFields,
+  sortFormErrors,
+  useUser,
+} from "utils";
+import { FormJson, FormField, UserRoles } from "types";
 
 export const Form = ({ id, formJson, onSubmit, children, ...props }: Props) => {
   const { fields, options } = formJson;
   const { reportData } = useContext(ReportContext);
-
   const formSchema = yupSchema(formJson.validation || {});
 
+  // determine if fields should be disabled (based on admin roles )
+  const { userRole } = useUser().user ?? {};
+  const isAdminUser =
+    userRole === UserRoles.ADMIN ||
+    userRole === UserRoles.APPROVER ||
+    userRole === UserRoles.HELP_DESK;
+  const shouldDisableAllFields = isAdminUser && formJson.adminDisabled;
+
```

Comment:
Nit: Would maybe rename this something like `formInputDisabled` or `fieldInputDisabled` or something so when we use it below to determine if a resolver is set it reads a bit better.